src/tail.ts

```diff
@@ -116,6 +130,32 @@ export class Tail {
     if (this.midHandle) {
       this.midHandle.position = newMid;
     }
+    if (this.spec) {
+      this.spec.midpointX = this.mid.x!;
```

Comment:
Well, this is complicated. Ideally, we would like to guarantee that the bubble object spec should be re-persisted here, since the tailSpec is a field of the spec?